### ISSUE TITLE
Tab-completed filenames should be single-quoted, not backtick-quoted

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/nushell#11815]
### Describe the bug

When tab-completing a filename, backticks are used, which means globs are now auto-expanded for all filesystem commands that accept them.  This means that tab-completing a filename which includes a glob character will typically result in an error or other undesired results.

### How to reproduce

```Nushell
> mkdir repro-testing; cd repro-testing
> touch  `2024-02-09 Backup [2].xml`
> mkdir backups
> cp 2024
# press Tab and select the file from the menu
> cp `2024-02-09 Backup [2].xml` backups\
Error: nu::shell::file_not_found

  × File not found
   ╭─[entry #55:1:1]
 1 │ cp `2024-02-09 Backup [2].xml` backups\
   ·    ─────────────┬─────────────
   ·                 ╰── file not found
   ╰────

# Up-arrow and manually edit commandline to
> cp '2024-02-09 Backup [2].xml' backups\
# Works
```

### Expected behavior

My suggestion was going to be to automatically double-quote a filename on tab-completion, which would work for the above case.  However, that will break if you complete a filename which includes a parent directory (and in a strange way, currently, but I'll probably open another issue on that).  So single-quoted is preferable, I believe.

Alternatively:

* Continue to backtick-quote the filename, but run the tab-completed result through `str escape-glob` first.
* Double-quote the string, but properly escape backslashes in directory hierarchy (e.g. `"foo\\bar"`)

### Configuration

Reproduced under both Windows and Linux 0.90.1 - Does not occur on 0.89.0, at least for the commands I tested, due to the 0.90.1 glob-rule changes.

| key                | value                                                |
| ------------------ | ---------------------------------------------------- |
| version            | 0.90.1                                               |
| branch             |                                                      |
| commit_hash        | c2992d5d8b8541b64e9fc1440ed2917fc9b757f7             |
| build_os           | windows-x86_64                                       |
| build_target       | x86_64-pc-windows-msvc                               |
| rust_version       | rustc 1.73.0 (cc66ad468 2023-10-03)                  |
| rust_channel       | 1.73.0-x86_64-pc-windows-msvc                        |
| cargo_version      | cargo 1.73.0 (9c4383fb5 2023-08-26)                  |
| build_time         | 2024-02-07 02:25:54 +00:00                           |
| build_rust_channel | release                                              |
| allocator          | mimalloc                                             |
| features           | dataframe, default, extra, sqlite, trash, which, zip |
| installed_plugins  |                                                      |

### Additional context

I could imagine an *extreme corner-case* in which the current behavior results in a *valid* glob-pattern which might `rm` files which weren't tab-completed.  It almost seems that it would require too pathologic a naming system as to be of concern, though.

**Comment by NotTheDr01ds** [comment]
Side-note: This is not a contrived scenario.  I actually ran into it today when attempting to backup my Evernote notebooks (was already too expensive, but dang, that new price-hike ...).  The filename generated included the brackets which triggered the issue described above.

**Comment by fdncred** [comment]
It seems like we have to pick one or the other, backticks or single quotes. So, someone is not going to be happy no matter what we choose, I guess.

- backticks - "I don't like the globbing"
- single quotes - "I need globbing"
- double quotes - "I don't need escaping"

what's the right answer?

**Comment by NotTheDr01ds** [comment]
> * backticks - "I don't like the globbing"

Well, when selecting a tab-complete **file** from a menu, when would you *like* globbing? ;-)

**Comment by fdncred** [comment]
> Put differently, is there any situation in which a tab-completion should result in a glob?

Well, if you want to do something like 
```
ls `c:\program files\a[bc]d*`
```
Then yes, you'd want to glob once the path with a space is completed.

It really seems to me like we may need a user level opt-in or opt-out of globbing on some commands like `ls --no-glob a[bc]d`

**Comment by sophiajt** [comment]
@fdncred agreed, though it needs to be at the syntax level and not the flag level as we'd have to put that flag on any commands that work with filepaths.

**Comment by fdncred** [comment]
> @fdncred agreed, though it needs to be at the syntax level and not the flag level as we'd have to put that flag on any commands that work with filepaths.

Oh, I'd love to have something (nearly) transparent to the user. I just don't see what that could be. Hopefully I'm just not seeing the forest for the trees.

**Comment by NotTheDr01ds** [comment]
@fdncred  Your use-case of tab-completion for directory, followed by glob makes sense.  I think it's far less common than the reverse scenario, though, but I can't rule out that someone might prefer the backticks behavior due to it.

Sounds like an `$env.config.filename_quotes` or something like that, perhaps?

**Comment by fdncred** [comment]
i've never personally liked backticks but the way our quoting system works, sometimes you need them because on windows, sometimes double quotes are useless since they interpret escapes and you just can't use them. so if you need to have quotes in quotes, your only real option is backticks and single quotes.

This problem that i mention with windows would probably be resolved, at least partially, if we had a raw string format like python's triple double quotes or rusts r"" quotes.

**Comment by WindSoilder** [comment]
In auto-completion context, it's handled by this pr: https://github.com/nushell/nushell/pull/11766

Would you build on latest main and try it again?

**Comment by NotTheDr01ds** [comment]
> In auto-completion context, it's handled by this pr: #11766
> 
> Would you build on latest main and try it again?

Thanks!  Built and tested on Linux and it seems to be exactly as needed (well, except for @fdncred 's use case ;-).

I'm not set up to build on Windows at the moment, but if I figure out how to cross-compile, I'll do a quick test on Windows as well, but I can't imagine any problems.

Closing as fixed by #11766
